products/views.py

```diff
@@ -1,3 +1,51 @@
-from django.shortcuts import render
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
 
-# Create your views here.
+from .models          import Brand, Type, Product
+
+class BrandsView(View):
+    def get(self, request):
+        brands  = Brand.objects.all().order_by("id")
+
+        catalog = [{
+            "id"    : brand.id,
+            "name"  : brand.name,
+            "image" : brand.image
+        } for brand in brands]
+
+        return JsonResponse({"catalog": catalog}, status=200)
+
+class TypesView(View):
+    def get(self, request):
+        types   = Type.objects.all().order_by("id")
+
+        catalog = [{
+            "id"    : type.id,
+            "name"  : type.name,
+            "image" : type.image
+        } for type in types]
+
+        return JsonResponse({"catalog": catalog}, status=200)
+
+class ProductsView(View):
+    def get(self, request):
+        brand_id = request.GET.get("brand")
+        type_id  = request.GET.get("type")
+        filter   = request.GET.get("filter", "id")
+
+        if brand_id or type_id:
+            products = Product.objects.filter(Q(brand=brand_id) | Q(type=type_id)).order_by("id")
+
+        else:
+            products = Product.objects.all().order_by("id")
+
+        items = [{
+                "id"          : product.id,
+                "name"        : product.name,
+                "price"       : product.price,
+                "thumbnail"   : product.thumbnail,
+                "like_number" : product.like_number
+            } for product in products.order_by(filter)]
```

Comment:
```suggestion
        items = [{
                "id"          : product.id,
                "name"        : product.name,
                "price"       : product.price,
                "thumbnail"   : product.thumbnail,
                "like_number" : product.like_number
            } for product in Product.object.filter(q).order_by(filter)]
```
1. 조건문에서는 filter에 대한 조건만 만듭니다. 그리고 완성된 조건을 filter에 넣어줍니다. 만약에 어떠한 조건문도 통과하지 못 하면 빈 Q가 들어가게 되고 이는 all()과 같은 효과를 나타냅니다.
2. 만약에 filter에 order_by 할 수 없는 값이 들어 올 때는 어떻게 될까요? 
3. Q 객체 말고도 dictionary를 이용하여 filter의 인자를 결정하는 방법이 있습니다. 이거는 적용하라는 것은 아니고 알고 있으면 도음이 될 것 같아서 알려드립니다. 한 번 찾아보세요~